core/src/main/java/org/elasticsearch/action/index/IndexRequest.java

```diff
@@ -660,7 +668,7 @@ public void process(MetaData metaData, @Nullable MappingMetaData mappingMd, bool
     @Override
     public void readFrom(StreamInput in) throws IOException {
         super.readFrom(in);
-        type = in.readString();
+        type = in.readOptionalString();
```

Comment:
Indeed. And I will backport it. But I think its just a think to keep in mind for then.